### ISSUE TITLE
[Bug] [iOS] DatePicker & TimePicker inconsistency on iOS 14

### ISSUE THREAD

**Issue by mduchev** [xamarin/Xamarin.Forms#12258]
### Description
The DatePicker & TimePicker views are looking strangely on iOS 14. The TimePicker also doesn't have the keyboard visible, as it does on Apple's apps. For reference, see apps like Phone, Calendar, Reminders, etc.

### Steps to Reproduce
1. Create a new Xamarin.Forms solution
2. Add DatePicker or TimePicker views to the page's content
3. Run the app (see the attached project below)

### Expected Behavior
The picker elements to be rendered normally. 

### Actual Behavior
The picker elements are rendered very strangely (showing initially the selected value) and the TimePicker doesn't have the keyboard on as it does on other iOS apps. They are functional (working) though.

### Basic Information
- Version with issue: 4.8.0.1451
- Last known good version: N/A
- IDE: Visual Studio 2019 for Mac
- Affected Devices: iOS devices

### Screenshots
![Sep-25-2020 17-26-56](https://user-images.githubusercontent.com/8217099/94279562-f589b580-ff54-11ea-8f1c-e819a96e68b5.gif)

For reference with Apple's apps:
![IMG_2869](https://user-images.githubusercontent.com/8217099/94280480-21596b00-ff56-11ea-9f50-718471b92863.PNG)
![IMG_2870](https://user-images.githubusercontent.com/8217099/94280490-25858880-ff56-11ea-93ca-62cec6d2ebdb.PNG)

### Reproduction Link
[DemoIos14Issues.zip](https://github.com/xamarin/Xamarin.Forms/files/5283397/DemoIos14Issues.zip)

**Comment by stesvis** [comment]
Same here, looks quite ugly.

**Comment by ChristianFrom** [comment]
I have the same issue on my application.

**Comment by stesvis** [comment]
Turns out that for the moment we can use a renderer and set the `PreferredDatePickerStyle` property.
This is what I did:

```cs
    public class MyDatePickerRenderer : DatePickerRenderer
    {
        protected override void OnElementChanged(ElementChangedEventArgs<DatePicker> e)
        {
            base.OnElementChanged(e);

            if (e.NewElement != null && this.Control != null)
            {
                try
                {
                    if (UIDevice.CurrentDevice.CheckSystemVersion(13, 2))
                    {
                        UIDatePicker picker = (UIDatePicker)Control.InputView;
                        picker.PreferredDatePickerStyle = UIDatePickerStyle.Wheels;
                    }
                }
                catch (Exception)
                {
                    // do nothing
                }
            }
        }
```

So basically for iOS 13.2+ force it to use the old wheel style. I put it in a try catch to use the new default if anything fails...

**Comment by mduchev** [comment]
The workaround workes for both Date- & TimePickers. Just FYI - CheckSystemVersion is greater than or equal to the specified major and minor values, so the correct check will be **CheckSystemVersion(14, 0)**. (ref: [docs](https://docs.microsoft.com/en-us/dotnet/api/uikit.uidevice.checksystemversion?view=xamarin-ios-sdk-12))

**Comment by stesvis** [comment]
Yes, I just check for 13.2 because i read that they started to introduce these changes in 13.2 (even tho i've seen it with 14+).
Either way, if i am wrong it might throw an exception and do nothing.

**Comment by wislon** [comment]
FYI: I've just tried it on iOS 13.5 and iOS 14.0. 

On 13.5, it's the original 'Wheels' implementation, but on iOS 14, it (without @stesvis' fix), it's the new implementation. 
The fix works perfectly on iOS 14, thanks so much @stesvis!  :)

**Comment by hartez** [comment]
See also #11963.

This should be addressed by #12288 and #12158 (which are similar to the fix from @stesvis upthread).

There's also #12139 in the works, which will allow you to set a UIDatePickerStyle from a platform specific.

**Comment by Serena247Pro** [comment]
The workaround did change the style perfectly. But it didn't help show the selected date? I also tried adding UITextField entry = Control;UIDatePicker picker = (UIDatePicker)entry.InputView; It doesn't work. Anything missing?

**Comment by Suplanus** [comment]
It's not the best way, but I create a new issue. Perhaps now someone is looking into it: #13373

**Comment by mduchev** [comment]
I can also confirm that it was working fine in 4.8.0.1821 and it's broken again in 5.0

**Comment by martinsmithdev** [comment]
Still an issue. Workaround works but needs fix.

**Comment by Suplanus** [comment]
Is fixed in this release: [5.0.0 Service Release 1](https://github.com/xamarin/Xamarin.Forms/releases/tag/release-5.0.0-sr1)

**Comment by Maxima078** [comment]
Is there a way to use compact mode but without the two steps before calendar appears ? Such as date picker within Safari.
